src/praisonai-agents/praisonaiagents/agent/agent.py

```
@@ -1937,7 +1937,181 @@ def run(self):
+                if isinstance(tool_param, MCP) and hasattr(tool_param, 'to_openai_tool'):
+                    logging.debug("Converting MCP tool to OpenAI format")
+                    openai_tool = tool_param.to_openai_tool()
+                    if openai_tool:
+                        if isinstance(openai_tool, list):
+                            tool_param = openai_tool
+                        else:
+                            tool_param = [openai_tool]
+                        logging.debug(f"Converted MCP tool: {tool_param}")
+
+            # Store chat history length for potential rollback
+            chat_history_length = len(self.chat_history)
```

Ruff (0.12.2)

Local variable `chat_history_length` is assigned to but never used

Remove assignment to unused variable `chat_history_length`

(F841)